tdrs-backend/tdpservice/reports/models.py

```diff
@@ -0,0 +1,116 @@
+"""Define report models."""
+
+from django.db import models
+from django.db.models import Max
+
+from ..stts.models import STT
+from ..users.models import User
+
+
+# The Report File model was starting to explode, and I think that keeping this logic
+# in its own abstract class is better for documentation purposes.
+class File(models.Model):
+    """Abstract type representing a file stored in S3."""
+
+    class Meta:
+        """Metadata."""
+
+        abstract = True
+
+    original_filename = models.CharField(max_length=256,
+                                         blank=False,
+                                         null=False)
+
+    slug = models.CharField(max_length=256, blank=False, null=False)
+    extension = models.CharField(max_length=8, default="txt")
+
```

Comment:
@riatzukiza Just for some context, do you mind sharing what the `original_filename`, `slug`, and `extension` fields will be used for?

Reply:
original_filename is remembered in our database because we won't be keeping the same filename in s3, and the designs suggest that the frontend expects files to be identified by their original names.

slug is what we call the file in s3, it'll be a UUID string generated by the frontend.

The extention is there because we are told that not all files have extentions on their file names, and so they get provided seperately from the name of the file by the frontend when thats the case.

Reply:
@riatzukiza These would make great code comments in the model file, I'm sure other devs looking back at this code would have similar questions to the ones @hursey013 asked.